frontend/src/components/RegionMap.js

```diff
@@ -1,16 +1,16 @@
 import React, {useEffect, useRef} from 'react';
 import maplibregl from 'maplibre-gl';
-import {useRegion} from "./RegionContext";
+import {useNavigation} from "./NavigationContext";
 import {fetchRegionGeometry} from "../api";
 
 const MapComponent = () => {
     const mapContainer = useRef(null);
     const map = useRef(null);
-    const { selectedRegion } = useRegion();
+    const { selectedRegion, selectedHierarchy } = useNavigation();
 
     const fetchSelectedRegionGeometry = async () => {
         if (selectedRegion.id !== null && selectedRegion.id !== 0) {
-            const response = await fetchRegionGeometry(selectedRegion.id);
+            const response = await fetchRegionGeometry(selectedRegion.id, selectedHierarchy.hierarchyId);
             if (response) {
                 return response.geometry;
             } else {
```

Comment:
> The useEffect hook is currently only dependent on selectedRegion. However, since the map initialization logic within the hook also depends on selectedHierarchy, it should be included in the dependency array to ensure that the map is re-initialized when the hierarchy changes.

Ok, I'll update the dependencies.

> The function fetchSelectedRegionGeometry lacks error handling for the asynchronous call to fetchRegionGeometry. 

I don't want to do it now. But I need to think about error handling for all API calls. Suggest a GH issue description so I can create the issue and not forget about it.

Reply:
#86 created.